oauth2.gemspec

```diff
@@ -4,7 +4,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'oauth2/version'
 
 Gem::Specification.new do |spec|
-  spec.add_dependency 'faraday', ['>= 0.8', '< 0.10']
+  spec.add_dependency 'faraday', ['>= 0.8', '~> 0.10']
```

Comment:
First of all, that's different from what the title of this PR says :)

Also you cannot do that since faraday does not promise backward compatibility. https://github.com/lostisland/faraday/blob/master/CONTRIBUTING.md says it will follow semver *once* it hits 1.0. So you should use `< 0.11`.